api/v1alpha1/fenceagentsremediation_webhook_test.go

```diff
@@ -11,58 +11,118 @@ var _ = Describe("FenceAgentsRemediation Validation", func() {
 
 	Context("creating FenceAgentsRemediation", func() {
 
-		When("agent name match format and binary", func() {
-			It("should be accepted", func() {
-				far := getTestFAR(validAgentName)
-				_, err := far.ValidateCreate()
-				Expect(err).ToNot(HaveOccurred())
+		Context("with ResourceDeletion strategy", func() {
+
+			When("agent name match format and binary", func() {
+				It("should be accepted", func() {
+					far := getTestFAR(validAgentName)
+					Expect(far.ValidateCreate()).Error().NotTo(HaveOccurred())
+				})
+			})
+
+			When("agent name was not found ", func() {
+				It("should be rejected", func() {
+					far := getTestFAR(invalidAgentName)
+					Expect(far.ValidateCreate()).Error().To(MatchError(ContainSubstring("unsupported fence agent: %s", invalidAgentName)))
+				})
 			})
 		})
 
-		When("agent name was not found ", func() {
-			It("should be rejected", func() {
-				far := getTestFAR(invalidAgentName)
-				_, err := far.ValidateCreate()
-				Expect(err).To(HaveOccurred())
-				Expect(err.Error()).To(ContainSubstring("unsupported fence agent: %s", invalidAgentName))
+		Context("with OutOfServiceTaint strategy", func() {
+			var outOfServiceStrategy *FenceAgentsRemediation
+
+			BeforeEach(func() {
+				orgValue := isOutOfServiceTaintSupported
+				DeferCleanup(func() { isOutOfServiceTaintSupported = orgValue })
+
+				outOfServiceStrategy = getFAR(validAgentName, OutOfServiceTaintRemediationStrategy)
+			})
+			When("out of service taint is supported", func() {
+				BeforeEach(func() {
+					isOutOfServiceTaintSupported = true
+				})
+				It("should be allowed", func() {
+					Expect(outOfServiceStrategy.ValidateCreate()).Error().NotTo(HaveOccurred())
+				})
+			})
+			When("out of service taint is not supported", func() {
+				BeforeEach(func() {
+					isOutOfServiceTaintSupported = false
+				})
+				It("should be denied", func() {
+					Expect(outOfServiceStrategy.ValidateCreate()).Error().To(MatchError(ContainSubstring(outOfServiceTaintUnsupportedMsg)))
+				})
 			})
 		})
 	})
 
 	Context("updating FenceAgentsRemediation", func() {
-		var oldFAR *FenceAgentsRemediation
-		When("agent name match format and binary", func() {
-			BeforeEach(func() {
-				oldFAR = getTestFAR(invalidAgentName)
+		Context("with ResourceDeletion strategy", func() {
```

Comment:
same as above, the tests inside do not depend on strategy

Reply:
Fixed it.